### PR TITLE
Implement task archiving

### LINKED ISSUE #60: feat(tasks): archiving and unarchiving tasks
# User Story

> As a novice user, I can archive tasks that are done, so that I can refer to them in future.

~ _tasks/deadlines, no. 14_

## Proposed Command Syntax

If not `cd`’d into module:
```
archive -c <module code> -t <task index>
unarchive -c <module code> -t <task index>
```

If `cd`’d into module:
```
archive -t <task index>
unarchive -t <task index>
```

For ease of implementation, the current selected module will be implicitly passed with the `-m` flag.

To show only unarchived tasks:
```
ls
```

To show all tasks, including archived tasks:
```
ls -A
```

### PR DESCRIPTION
**Pull request by jontmy**
## Tasks

- [x] `ls -A` command to show all tasks, including those hidden
- [x] `ls` command to show only unarchived tasks
- [x] `archive <index>` command to archive a task
- [x] `unarchive <index>` command to unarchive a task

For more information, see #60.

### CODE REVIEW

**Comment by jontmy** [comment]
**Demonstration of `ls` and `ls -A` working:**

https://user-images.githubusercontent.com/50949210/194843890-a2ec4374-724e-4538-bb4d-26da88ddf852.mov

**Comment by jontmy** [comment]
**Working demo:**

https://user-images.githubusercontent.com/50949210/195113310-a41f66ce-ef51-4835-bf23-38bf1f56e6f6.mp4

File: src/main/java/modtrekt/logic/parser/util/CommandParser.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+ * Unlike ArgumentTokenizer, CommandParser allows validation of required prefixes, as well as
+ * specifying optional prefixes and expected usage message with a fluent API.
+ * Additionally, CommandParser also exposes its utility methods used to validate the arguments of a command.
+ * <br>
+ * The usage message is printed whenever a required prefix is missing via the ParseException that is thrown,
+ * along with a list of the missing prefixes.
+ * <br>
+ * Use CommandParserBuilder to create a CommandParser with a fluent API.
+ */
+public class CommandParser {
```
Comment (domoberzin) [inline src/main/java/modtrekt/logic/parser/util/CommandParser.java:28]: Good job adding this! Think we can integrate this with our other command parsers

**Review by domoberzin — COMMENTED** [review]
LGTM, great work adding the CommandParser, think we should standardize to use it for all our commands.

Also, are we going to support archiving without specifying a module? i.e. Are we supporting archive -t <task index as seen on list> as a command? Cause I'm not exactly sure what utility the module serves, other than validating the task and module association, since we can go off of the task index, as we already linked each task to a specific module.

Anyway we should probably discuss that another time, think its good to merge! Once again nice work, think using the CommandParser will make all our parser classes more clean

**Review by domoberzin — APPROVED** [review]

**Pull request merged**